From 04d2895125d9748b5d19f42ddddec83d3896d635 Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@canonical.com>
Date: Wed, 7 May 2014 11:52:31 +0100
Subject: [PATCH] Non-maintainer upload.

* Non-maintainer upload.
* Source profile.d from /etc/bash.bashrc to allow packages ship
  integration hooks into interactive shells, same ones they already ship
  for login shells. (Closes: #675008)
---
 debian/changelog       |  9 +++++++++
 debian/etc.bash.bashrc | 22 ++++++++++++++++++++++
 2 files changed, 31 insertions(+)

diff --git a/debian/changelog b/debian/changelog
index ccda677..6bc7729 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,3 +1,12 @@
+bash (4.3-7.1) UNRELEASED; urgency=medium
+
+  * Non-maintainer upload.
+  * Source profile.d from /etc/bash.bashrc to allow packages ship
+    integration hooks into interactive shells, same ones they already ship
+    for login shells. (Closes: #675008)
+
+ -- Dimitri John Ledkov <xnox@ubuntu.com>  Wed, 07 May 2014 10:50:11 +0100
+
 bash (4.3-7) unstable; urgency=medium
 
   * Apply upstream patches 009 - 011 (replacing local patches):
diff --git a/debian/etc.bash.bashrc b/debian/etc.bash.bashrc
index 54687b1..e38b202 100644
--- a/debian/etc.bash.bashrc
+++ b/debian/etc.bash.bashrc
@@ -53,3 +53,25 @@ if [ -x /usr/lib/command-not-found -o -x /usr/share/command-not-found/command-no
 		fi
 	}
 fi
+
+# Source profile.d even though "it's meant to be for login shells
+# only", because we need a hook integration for
+# e.g. command-not-found, bash-completion, gnome-terminal/vte,
+# screen/byobu etc. And e.g. on fedora systems, bashrc does source
+# profile.d. If something does break, things in /etc/profile.d will
+# need to differentiate interractive shells by inspecting "$PS1" (or
+# similar as /etc/profile.d/vte.sh already does). If there is bad
+# interraction between snippets in profile.d, that's already a bug,
+# against affected packages to sort out, since login shells are
+# already affected.
+#
+# Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=675008
+
+if [ -d /etc/profile.d ]; then
+  for i in /etc/profile.d/*.sh; do
+    if [ -r $i ]; then
+      . $i
+    fi
+  done
+  unset i
+fi
-- 
2.0.0.rc0

